### ISSUE TITLE
Placeholder text colour contrast doesn't meet the accessibility standards

### ISSUE THREAD

**Issue by AlicjaSuska** [sourcegraph/sourcegraph#11963]
Currently used placeholder text color (`#2B3750` 50% opacity) doesn't meet the accessibility standards. My proposition is to change it to ~~`#6C7584` 100%~~ `#2b3750` 70% opacity. This is the closest color to the currently used one, that meets the standards.

Placeholder text is used in many places:
- main search bar ghost text 'Search code'
- query builder text fields
- site admin - user stats search bar
- text fields in profile settings etc.

See the [document](https://www.figma.com/file/f8jfzPiPr0MiGCZ1nTsIoi/Placeholder-text-color?node-id=0%3A1) that showcases current color and new proposition with contrast ratio test results.

@rrhyne @christinaforney @felixfbecker Please, review. Thank you!

**Comment by felixfbecker** [comment]
@AlicjaSuska could you choose one from our [greyish color palette](https://github.com/sourcegraph/sourcegraph/blob/763e4078ba3fa6cd937027a006af9356456ebb52/shared/src/global-styles/colors.scss#L6-L31)?

**Comment by felixfbecker** [comment]
Also don't forget to check the contrast in dark theme too :)

**Comment by AlicjaSuska** [comment]
Hi @felixfbecker, thank you for the feedback. When it comes to choosing from the existing palettes:
- grey-ish palette becomes too blue to use it directly for lighter tones. The text starts looking like a link, not like a placeholder
- grey from the basic palette - grey 5 doesn't meet the standards, grey 7 is too dark (it looks like a regular text).

We can:
- increase the opacity of the currently used color from 50% to 70% to meet the accessibility standards.
or
- introduce the new color to the palette #6B7385 for placeholder texts.

In general, it is a bigger discussion about if we would like to use the opacity for at all. It limits the number of colors we need to define but, in reality, we create even more inconsistencies and cases as the element's color depends on the background. we've discussed it with @felixfbecker briefly. I think we've agreed that no opacity would be a better solution. @rrhyne what is your take on that?

**Comment by felixfbecker** [comment]
> grey from the basic palette - grey 5 doesn't meet the standards, grey 7 is too dark (it looks like a regular text).

Since you mentioned 5 and 7, what about open color 6 (`#868e96`)?

We also need to pick one for dark themes.

**Comment by AlicjaSuska** [comment]
For the dark theme, we can use Grey 6 (#878E95). It is very similar to the color we're using now. Also, it passes the test.

For the light theme:
Grey 6 doesn't meet the accessibility standards. Grey 7 is too dark - it looks like a regular text. Here is the comparison:
<img width="1513" alt="Screenshot 2020-07-12 at 11 39 00" src="https://user-images.githubusercontent.com/20326070/87243241-b8715500-c434-11ea-830d-57a0fe3ab91c.png">

It looks like for the light theme, we need to introduce another color. If we decide to use 100% opacity, I would go with #6B7385 (effectively the same as currently used grey #2b3750 with 70% opacity on the white background).

@felixfbecker

**Comment by rrhyne** [comment]
I agree, we should not use opacity to affect color/brightness.

**Comment by AlicjaSuska** [comment]
@felixfbecker @rrhyne can we agree on the following and move forward with those changes:

- dark theme - Grey 6 (#878E95)
- light theme - add new color #6B7385 (effectively the same as currently used grey #2b3750 with 70% opacity on the white background).

Please, share your feedback. Thank you!

**Comment by felixfbecker** [comment]
I would generally try to avoid alpha channels too to constraint the number of colors, but if that means we need to add a color to the blueish scale just for this, I think I would prefer to just make an exception here and use the alpha channel. Inserting a new color would mean I have to assign it an index (we currently have exactly 25 steps) and re-assign all following colors (and change all references step-by-step) so the number stays incremental. But trying to find the ideal index to insert it at, it's quite clear the color doesn't really fit into the sequence at all. Brightness-wise I would probably put it between 15 and 16, but it's really out of place because all other adjacent colors have 30% saturation, while this one has 11% (and it's visually noticable).

![2020-07-15 17 48 52](https://user-images.githubusercontent.com/10532611/87566889-e3bda380-c6c3-11ea-9652-914627445eba.gif)

So while I'd like to avoid the alpha where possible, I'd like to make an exception here.
